metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationZkWriter.java

```diff
@@ -146,6 +147,15 @@ void handleTopicsSnapshot(TopicsImage topicsImage, KRaftMigrationOperationConsum
         Map<Uuid, Map<Integer, PartitionRegistration>> changedPartitions = new HashMap<>();
         Map<Uuid, Map<Integer, PartitionRegistration>> newPartitions = new HashMap<>();
 
+        Set<String> pendingTopicDeletions = migrationClient.topicClient().readPendingTopicDeletions();
+        if (!pendingTopicDeletions.isEmpty()) {
+            operationConsumer.accept(
+                DELETE_PENDING_TOPIC_DELETION,
+                "Delete pending topic deletions",
+                migrationState -> migrationClient.topicClient().clearPendingTopicDeletions(pendingTopicDeletions, migrationState)
```

Comment:
hmm, I think this logic to clear pending topic deletions should exist in the code to sync kraft state to zk state rather than here, right? I think the other stuff here just reads and doesn't write (correct?)

Reply:
Yea, this is in `handleTopicsSnapshot` which is sync'ing the TopicImage to ZK. Really it doesn't need to happen each time when we handle a snapshot, but I figured putting it here was better than having additional one-off logic at migration time